distributed/protocol/tests/test_torch.py

```diff
@@ -14,15 +14,17 @@ def test_tensor():
     assert (x == t2.numpy()).all()
 
 
-def test_grad():
+@pytest.mark.parametrize("requires_grad", [True, False])
+def test_grad(requires_grad):
     x = np.arange(10)
-    t = torch.Tensor(x)
+    t = torch.tensor(x, dtype=torch.float, requires_grad=requires_grad)
     t.grad = torch.zeros_like(t) + 1
 
     t2 = deserialize(*serialize(t))
-    assert (t2.numpy() == x).all()
-    assert (t2.grad.numpy() == 1).all()
 
+    assert (t2.detach().numpy() == x).all()
+    assert (t2.grad.numpy() == 1).all()
+    assert (t2.requires_grad is requires_grad)
```

Comment:
I'd add a test to make sure that `t.requires_grad` isn't modified by serialization:

``` diff
+     assert t.requires_grad is requires_grad
```

Reply:
And maybe use `np.allclose` instead of `==`?